include/verilated_types.h

```diff
@@ -1238,6 +1238,40 @@ static inline bool VL_CAST_DYNAMIC(VlClassRef<T> in, VlClassRef<U>& outr) {
     }
 }
 
+//=============================================================================
+// VlSampleQueue stores samples for input clockvars in clocking blocks. At a clocking event,
+// samples from this queue should be written to the correct input clockvar.
+
+template <typename T_Sampled>
+class VlSampleQueue final {
+    // TYPES
+    // Type representing a single value sample at a point in time
+    struct VlSample {
+        uint64_t m_timestamp;  // Timestamp at which the value was sampled
+        T_Sampled m_value;  // The sampled value
+    };
+
+    // MEMBERS
+    std::vector<VlSample> m_queue;  // Queue of samples with timestamps
```

Comment:
Given the find_if and insert below, is vector the best data type here?

Reply:
Made it a deque, and now `pop` does a binary search on it, as it should always be sorted.